### PR TITLE
Add help for ~:[s]-[e]   show lines s-e & fix #8164 recursive help

### LINKED ISSUE #8164: Recursive help fail
```
[0x00000000]> w?*~ijn
| wop[DO] [arg]                 De Bruijn Patterns
[0x00000000]> wo?*~ijn
| wop[DO] [arg]                 De Bruijn Patterns
[0x00000000]> ?*~ijn
[0x00000000]>
```

### PR DESCRIPTION
**Pull request by MaskRay**
#8461

### CODE REVIEW

**Comment by Maijin** [comment]
Undefined reference - https://travis-ci.org/radare/radare2/jobs/274023339#L859

**Comment by MaskRay** [comment]
Makefile cheated me.

Moved r_core_cmd_help features to r_cons

File: libr/cons/cons.c [hunk 1]
```diff
@@ -1322,3 +1322,50 @@ R_API void r_cons_breakword(const char *s) {
...
+ * "command2", "args2", "description"}; */
+R_API void r_cons_cmd_help(const char *help[], bool use_color) {
+	int i, max_length, padding_length;
+	RCons *cons = r_cons_singleton ();
+	char const *args_color_start;
+	char const *help_color_start;
+	char const *reset_colors;
+	char padding[256];
+
+	args_color_start = use_color ? cons->pal.args: "";
```
Comment (radare) [inline libr/cons/cons.c]: put that assignment in the same definition line or put all 3 assignments in a if (use_color) { block

Comment (radare) [reply libr/cons/cons.c]: better use different varnames for storing the colors. like "pal_" + ...

File: libr/cons/cons.c [hunk 2]
```diff
@@ -1322,3 +1322,50 @@ R_API void r_cons_breakword(const char *s) {
...
+	help_color_start = use_color ? cons->pal.help: "";
+	reset_colors     = use_color ? cons->pal.reset: "";
+
+	max_length = 0;
+	for (i = 0; help[i]; i += 3) {
+		int len0 = strlen (help[i]);
+		int len1 = strlen (help[i+1]);
+		if (i) max_length = R_MAX (len0+len1, max_length);
```
Comment (radare) [inline libr/cons/cons.c]: missing braces and spaces around +

File: libr/cons/cons.c [hunk 3]
```diff
@@ -1322,3 +1322,50 @@ R_API void r_cons_breakword(const char *s) {
...
+	}
+
+	for (i = 0; help[i]; i += 3) {
+		padding_length = max_length - \
+			(strlen (help[i]) + strlen (help[i+1]));
```
Comment (radare) [inline libr/cons/cons.c]: missing space around +

File: libr/cons/cons.c [hunk 4]
```diff
@@ -1322,3 +1322,50 @@ R_API void r_cons_breakword(const char *s) {
...
+		if (padding_length >0) {
+			memset (padding, ' ', padding_length);
+			padding[padding_length] = '\0';
+		} else padding[0] = 0;
```
Comment (radare) [inline libr/cons/cons.c]: missing braces

File: libr/cons/cons.c [hunk 5]
```diff
@@ -1322,3 +1322,50 @@ R_API void r_cons_breakword(const char *s) {
...
+	char const *help_color_start;
+	char const *reset_colors;
+	char padding[256];
+
+	args_color_start = use_color ? cons->pal.args: "";
+	help_color_start = use_color ? cons->pal.help: "";
+	reset_colors     = use_color ? cons->pal.reset: "";
+
+	max_length = 0;
+	for (i = 0; help[i]; i += 3) {
```
Comment (radare) [inline libr/cons/cons.c]: may crash if not checking null on +1 and +2

**Comment by MaskRay** [comment]
i just moved the function to another file...

File: libr/cons/cons.c (hunk 5 above)
Comment (MaskRay) [reply libr/cons/cons.c]: I know it'll crash, but with the current code you can catch when help messages were not written correctly.

File: libr/cons/cons.c (hunk 4 above)
Comment (MaskRay) [reply libr/cons/cons.c]: Removed

File: libr/cons/cons.c (hunk 3 above)
Comment (MaskRay) [reply libr/cons/cons.c]: Rewritten

File: libr/cons/cons.c (hunk 2 above)
Comment (MaskRay) [reply libr/cons/cons.c]: Done

**Comment by MaskRay** [comment]
Rewritten

**Pull request merged**